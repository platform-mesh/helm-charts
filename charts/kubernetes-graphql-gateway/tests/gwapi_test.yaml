suite: Test Kubernetes Gateway API
values:
- ../values.yaml
- ../test_values.yaml
release:
  name: platform-mesh
  namespace: platform-mesh-system
chart:
  appVersion: 1.0.0
templates:
  - gwapi-referencegrant.yaml
  - gwapi-httproute.yaml
  - istio-authorizationpolicy.yaml
  - istio-destinationrule.yaml
  - istio-peerauthentication.yaml
  - istio-requestauthentication.yaml
  - istio-virtualservice.yaml
tests:
  - it: enable Istio, disable Gateway API - match the Snapshot
    set:
      istio.enabled: true
      gwApi.enabled: false
    asserts:
    - matchSnapshot: {}
  - it: disable Istio, enable Gateway API - match the Snapshot
    set:
      istio.enabled: false
      gwApi.enabled: true
    asserts:
      - matchSnapshot: {}