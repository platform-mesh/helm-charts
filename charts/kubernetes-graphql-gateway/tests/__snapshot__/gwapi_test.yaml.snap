disable Istio, enable Gateway API - match the Snapshot:
  1: |
    apiVersion: gateway.networking.k8s.io/v1
    kind: HTTPRoute
    metadata:
      name: platform-mesh-kubernetes-graphql-gateway
      namespace: default
    spec:
      hostnames:
        - portal.dev.local
        - '*.portal.dev.local'
      parentRefs:
        - name: k8sapi-gateway
          sectionName: websecure
      rules:
        - backendRefs:
            - name: kubernetes-graphql-gateway
              namespace: platform-mesh-system
              port: 8080
          filters:
            - type: URLRewrite
              urlRewrite:
                path:
                  replacePrefixMatch: /
                  type: ReplacePrefixMatch
            - requestHeaderModifier:
                set:
                  - name: Host
                    value: portal.dev.local
              type: RequestHeaderModifier
          matches:
            - path:
                type: PathPrefix
                value: /api/kubernetes-graphql-gateway/
  2: |
    apiVersion: gateway.networking.k8s.io/v1beta1
    kind: ReferenceGrant
    metadata:
      name: platform-mesh-kubernetes-graphql-gateway
      namespace: platform-mesh-system
    spec:
      from:
        - group: gateway.networking.k8s.io
          kind: HTTPRoute
          namespace: default
      to:
        - group: ""
          kind: Service
          name: platform-mesh-kubernetes-graphql-gateway
enable Istio, disable Gateway API - match the Snapshot:
  1: |
    apiVersion: networking.istio.io/v1alpha3
    kind: DestinationRule
    metadata:
      name: platform-mesh-kubernetes-graphql-gateway
      namespace: platform-mesh-system
    spec:
      host: platform-mesh-kubernetes-graphql-gateway.platform-mesh-system.svc.cluster.local
      trafficPolicy:
        tls:
          mode: ISTIO_MUTUAL
  2: |
    apiVersion: security.istio.io/v1beta1
    kind: PeerAuthentication
    metadata:
      name: platform-mesh-kubernetes-graphql-gateway
      namespace: platform-mesh-system
    spec:
      mtls:
        mode: STRICT
      selector:
        matchLabels:
          app: platform-mesh-kubernetes-graphql-gateway
  3: |
    apiVersion: networking.istio.io/v1alpha3
    kind: VirtualService
    metadata:
      name: platform-mesh-kubernetes-graphql-gateway
      namespace: platform-mesh-system
    spec:
      gateways:
        - platform-mesh-system/gateway
      hosts:
        - '*'
      http:
        - corsPolicy:
            allowHeaders:
              - Authorization
              - Content-Type
              - '*'
            allowMethods:
              - GET
              - POST
            allowOrigins:
              - regex: .*
          match:
            - uri:
                prefix: /kubernetes-graphql-gateway/
          rewrite:
            uri: /
          route:
            - destination:
                host: platform-mesh-kubernetes-graphql-gateway.platform-mesh-system.svc.cluster.local
                port:
                  number: 8080
